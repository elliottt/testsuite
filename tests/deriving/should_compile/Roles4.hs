{-# LANGUAGE RoleAnnotations #-}

module Roles4 where

class C1 a@N where
  meth1 :: a -> a

class C2 a@R where
  meth2 :: a -> a

type Syn1 a@N = [a]

class C3 a where
  meth3 :: a -> Syn1 a

