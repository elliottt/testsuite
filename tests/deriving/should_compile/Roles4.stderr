TYPE SIGNATURES
TYPE CONSTRUCTORS
  C1 :: * -> Constraint
  class C1 a@N
      RecFlag NonRecursive
      meth1 :: a -> a
  C2 :: * -> Constraint
  class C2 a@R
      RecFlag NonRecursive
      meth2 :: a -> a
  C3 :: * -> Constraint
  class C3 a@N
      RecFlag NonRecursive
      meth3 :: a -> Syn1 a
  Syn1 :: * -> *
  type Syn1 a@N = [a]
COERCION AXIOMS
  axiom Roles4.NTCo:C1 :: C1 a = a -> a
  axiom Roles4.NTCo:C2 :: C2 a = a -> a
  axiom Roles4.NTCo:C3 :: C3 a = a -> Syn1 a
Dependent modules: []
Dependent packages: [base, ghc-prim, integer-gmp]

==================== Typechecker ====================

